javabuilder-authorizer/lambda_function.rb

```diff
@@ -0,0 +1,78 @@
+require 'aws-sdk-lambda'
+require 'jwt'
+
+# The lambda handler takes an event with the header
+# 'Authorization: Bearer <token>',  where the token is a JWT token
+# generated by dashboard. The handler verifies the token with the appropriate
+# public key (dependant on the environment the request came from),
+# and checks the token has not expired and its issue time is not in the future.
+def lambda_handler(event:, context:)
+  headers = event['headers']
+  auth_header = headers['Authorization']
+  method_arn = event['methodArn']
+
+  if auth_header.start_with?('Bearer: ')
+    jwt_token = auth_header.split[1]
+    begin
+      decoded_token = JWT.decode(
+        jwt_token,
+        get_public_key(method_arn),
+        true,
+        verify_iat: true, # verify issued at time is valid
+        algorithm: 'RS256'
+      )
+    rescue JWT::ExpiredSignature, JWT::InvalidIatError
+      return generate_deny(nil, method_arn)
+    end
+
+    return generate_deny(nil, method_arn) unless decoded_token
+
+    generate_allow('me', method_arn, decoded_token[0])
```

Comment:
Could you leave a comment for `'me'` here? i.e. I'm unsure if it's a placeholder or a value with meaning.

Reply:
good callout. I had copied that from the sample code and hadn't thought about it since. I tested and it seems we can put anything here, so I'm thinking I will change this to user id.

Reply:
updated to use environment variables. It's slightly less ideal because the newlines aren't preserved in an environment variable, so I have to find/replace the escaped newline with a real one.